src/Tools/SchemaTool.php

```diff
@@ -83,6 +86,12 @@ public function createSchema(array $classes): void
                 throw ToolsException::schemaToolFailure($sql, $e);
             }
         }
+
+        $eventManager = $this->em->getEventManager();
+        $eventManager->dispatchEvent(
+            ToolEvents::postSchemaChanged,
+            new SchemaChangedEventArgs($this->em, $schema, new Schema(), $createSchemaSql),
```

Comment:
thank you for looking into this. so the idea would be that e.g. symfony messsenger could listen to this, inspect the changes and fire its own sql statements against the connection if it sees that something was done which it wants to alter?

is this event also useful when doctrine migrations are used rather than database updated on the fly?

with dbal 3, symfony was adding extra sql in https://github.com/symfony/symfony/blob/73f8713e143c64dc5f045dcdee7d5e893cccfcd8/src/Symfony/Bridge/Doctrine/SchemaListener/MessengerTransportDoctrineSchemaListener.php#L76C1-L76C38 (and [that sql](https://github.com/symfony/symfony/blob/73f8713e143c64dc5f045dcdee7d5e893cccfcd8/src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/PostgreSqlConnection.php#L115) afaik can not be represented in the doctrine schema models.
that sql got saved to the migration file, which was quite nice to review it.

Reply:
Yes, something like:

```php
public function onPostSchemaChanged($event)
{
    if (!$this->hasMessengerTable($event)) {
        $event->getEntityManage()->getConnection()->executeStatement($this->getMessengerTableSql());
    }
}
```

Reply:
and how can this be integrated with the doctrine migrations? could the listener also add sql into a migration?